lib/Runtime/Library/BuiltIn/BuiltIn.js

```diff
@@ -0,0 +1,60 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function (intrinsic) {
+    var platform = intrinsic.BuiltIn;
+
+    let FunctionsEnum = {
+        ArrayIndexOf: { className: "Array", methodName: "indexOf", argumentsCount: 1 },
+    };
+
+    platform.registerFunction(FunctionsEnum.ArrayIndexOf, function (searchElement, fromIndex) {
+        let o = __chakraLibrary.Object(this);
+        let len = __chakraLibrary.toLength(o["length"]);
+
+        if (len === 0) {
+            return -1;
+        }
+
+        let n = __chakraLibrary.toInteger(fromIndex);
+
+        if (n >= len) {
+            return -1;
+        }
+
+        let k;
+
+        if (n >= 0) {
+            if (len == -0) {
+                k = +0;
+            } else {
+                k = n;
+            }
+        } else if (n < 0) {
+            k = len + n;
+
+            if (k < 0) {
+                k = 0;
+            }
+        }
+
+        while (k < len) {
+            if (k in o) {
+                let elementK = o[k];
```

Comment:
Yes it is, we checked the behavior of this syntax in C++ and it is exactly the same.

---
In reply to: [115389001](https://github.com/Microsoft/ChakraCore/pull/2898#discussion_r115389001) [](ancestors = 115389001)